src/contracts/crowdfund.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  SmartContract,
+  method,
+  prop,
+  assert,
+  PubKeyHash,
+  PubKey,
+  SigHashPreimage,
+  Tx,
+  toHex,
+  Utils,
+  hash256,
+  SigHash,
+  Sig,
+  bsv,
+  UTXO,
+} from "scrypt-ts";
+
+export class Crowdfund extends SmartContract {
+  @prop()
+  recepient: PubKeyHash;
+
+  @prop()
+  contributor: PubKey;
+
+  @prop()
+  deadline: bigint;
+
+  @prop()
+  target: bigint;
+
+  constructor(
+    recepient: PubKeyHash,
+    contributor: PubKey,
+    deadline: bigint,
+    target: bigint
+  ) {
+    super(recepient, contributor, deadline, target);
+    this.recepient = recepient;
+    this.contributor = contributor;
+    this.deadline = deadline;
+    this.target = target;
+  }
+
+  @method()
+  public collect(txPreimage: SigHashPreimage, raisedAmount: bigint) {
+    assert(this.checkPreimage(txPreimage));
```

Comment:
You can use `this.ctx.hashOutputs` in the latest version without `SigHashPreimage`, like in https://scrypt.io/scrypt-ts/tutorials/stateful-contract.

Reply:
Don't understand that I am afraid. Seems like I need to use a stateful contract and this is not a stateful contract.